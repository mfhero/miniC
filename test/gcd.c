int gcd(int a, int b) {
    if (b == 0)
        return a;
    else 
        return gcd(b, a % b);
}

int main() {
    int n;
    n = getint();
    while (n > 0) {
        n = n - 1;
        int a;
        int b;
        a = getint();
        b = getint();
        int r;
        r = gcd(a, b);
        putint(r);
        putchar(10);
    }
    return 0;
}
