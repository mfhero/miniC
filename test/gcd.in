10
25 40
1342 123
12387 101
1001 49
121 12321
0 2
123 369
222 234
34 232
1 132410
