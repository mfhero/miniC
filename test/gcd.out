5
1
1
7
1
2
123
6
2
1
